gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxFormGroup,
+  NxToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon,
+  NxCode
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+import { themingEnabled, themeOverride, setThemingEnabled, setThemeOverride } from '../themeClassManager';
+
+type Props = {
+  onClose: () => void;
+};
+
+const DarkModeModal = ({ onClose }: Props) => {
+
+  const [themingEnabledState, setThemingEnabledState] = useState(themingEnabled),
+      [themeOverrideState, setThemeOverrideState] = useState(themeOverride);
+
+  useEffect(function() {
+    setThemingEnabled(themingEnabledState);
+  }, [themingEnabledState]);
+
+  useEffect(function() {
+    setThemeOverride(themeOverrideState);
+  }, [themeOverrideState]);
+
+  return (
+    <NxModal id="nx-modal-dark-mode-example"
+             onCancel={onClose}
+             aria-labelledby="nx-modal-dark-mode-header">
+      <header className="nx-modal-header">
+        <h2 className="nx-h2" id="nx-modal-dark-mode-header">
+          Setting Preferences for Light or Dark Mode
+        </h2>
+      </header>
+      <div className="nx-modal-content">
+        <NxFormGroup label="Enable Theme Changes">
+          <NxToggle onChange={() => setThemingEnabledState(!themingEnabledState)} isChecked={themingEnabledState}>
+            Opt-in to Allow Theming
+            <NxTooltip title={
+              <>Opting in will allow you to make adjustments to the display theme. Failure
+                to do so will always result in light mode. This operates by toggling the presence of
+                the <NxCode>nx-html--enable-color-schemes</NxCode> class.
+              </>}>
+              <NxFontAwesomeIcon className="dark-mode-modal__info-icon" icon={faInfoCircle}/>
```

Comment:
I wonder if this classname `dark-mode-modal__info-icon` is necessary here and below

Reply:
> I wonder if this classname `dark-mode-modal__info-icon` is necessary here and below

Thanks for catching that. I thought I needed it when initially building, and forgot to remove. Removed here: [4780ec8](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/4780ec883c67eeff90b0d1148ab0ff658756fb01)